### PR TITLE
Add parsing of the `tcbInfo` inner field

### PR DESCRIPTION
**Pull request by nick-mobilecoin**
Add parsing of the `tcbInfo` field of the retrieved TCB data from
<https://api.trustedservices.intel.com/sgx/certification/v4/tcb?fmspc={}>

### CODE REVIEW

**Comment by nick-mobilecoin** [comment]
Current dependencies on/for this PR:
* main
  * **PR #83** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/83" target="_blank">Graphite</a> 
    * **PR #84** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/84" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/84?utm_source=stack-comment).

File: verifier/src/tcb.rs [hunk 1]
```diff
@@ -25,11 +49,75 @@ impl From<serde_json::Error> for Error {
     }
 }
 
-/// Raw unverified representation of the TCB info provided from
+/// The `tcb_info` member of the TCB(Trusted Computing Base) data retrieved from
+/// <https://api.trustedservices.intel.com/sgx/certification/v4/tcb?fmspc={}>
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct TcbInfo {
```
Comment (nick-mobilecoin) [inline verifier/src/tcb.rs:56]: Up for suggestions on the names of these structs

File: verifier/src/tcb.rs [hunk 2]
```diff
@@ -25,11 +49,75 @@ impl From<serde_json::Error> for Error {
...
+    id: String,
+    version: u32,
+    issue_date: String,
+    next_update: String,
+    fmspc: String,
+    pce_id: String,
+    tcb_type: u32,
```
Comment (nick-mobilecoin) [inline verifier/src/tcb.rs:63]: Still trying to get to the full schema but intel's website has been a bit flaky today. I'm pretty sure I've seen a schema for what is in the json, for now I mainly based this off the example.json and the _live_ one I pulled

File: verifier/src/tcb.rs [hunk 3]
```diff
@@ -63,31 +152,188 @@ mod tests {
...
+    fn parse_example_tcb_info() {
+        let json = include_str!("../data/tests/example_tcb.json");
+        let tcb_raw = TcbInfoRaw::try_from(json).expect("Failed to parse raw TCB");
+        let tcb_info = TcbInfo::try_from(tcb_raw.tcb_info.get()).expect("Failed to parse TCB info");
+        assert_eq!(tcb_info.id, "SGX");
+        assert_eq!(tcb_info.version, 3);
+        assert_eq!(tcb_info.issue_date, "2022-04-13T09:38:17Z");
+        assert_eq!(tcb_info.next_update, "2022-05-13T09:38:17Z");
+        assert_eq!(tcb_info.fmspc, "50806F000000");
+        assert_eq!(tcb_info.pce_id, "0000");
```
Comment (nick-mobilecoin) [inline verifier/src/tcb.rs:193]: For this and the subsequent test I'm on the fence how much value the verification of the members provide, it feels like a wall of text trying to grok the test, but asserting that the try_from() worked felt a bit too loose.

**Review by cbeck88 — APPROVED** [review]

**Pull request merged**